<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="0" failed="4" ignored="0" total="5" passed="1">
  <reporter-output>
  </reporter-output>
  <suite name="Suite" duration-ms="65832" started-at="2019-03-01T04:42:34Z" finished-at="2019-03-01T04:43:40Z">
    <groups>
    </groups>
    <test name="Test" duration-ms="65832" started-at="2019-03-01T04:42:34Z" finished-at="2019-03-01T04:43:40Z">
      <class name="qa.sampleproject.tests.SampleTest">
        <test-method status="PASS" signature="setup()[pri:0, instance:qa.sampleproject.tests.SampleTest@77cd7a0]" name="setup" is-config="true" duration-ms="12035" started-at="2019-03-01T04:42:59Z" finished-at="2019-03-01T04:43:11Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="FAIL" signature="loginTest()[pri:3, instance:qa.sampleproject.tests.SampleTest@77cd7a0]" name="loginTest" duration-ms="21584" started-at="2019-03-01T04:43:11Z" finished-at="2019-03-01T04:43:33Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at qa.sampleproject.utilities.util.takesScreenshotAtEndofTest(util.java:27)
at qa.sampleproject.utilities.WebEventLister.onException(WebEventLister.java:145)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$0(EventFiringWebDriver.java:84)
at com.sun.proxy.$Proxy7.onException(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$1(EventFiringWebDriver.java:107)
at com.sun.proxy.$Proxy8.findElement(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.findElement(EventFiringWebDriver.java:194)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:69)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at com.sun.proxy.$Proxy12.sendKeys(Unknown Source)
at qa.sampleproject.pages.yahoologinpage.yahoologin(yahoologinpage.java:32)
at qa.sampleproject.tests.SampleTest.loginTest(SampleTest.java:32)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest -->
        <test-method status="FAIL" signature="teardown()[pri:0, instance:qa.sampleproject.tests.SampleTest@77cd7a0]" name="teardown" is-config="true" duration-ms="6877" started-at="2019-03-01T04:43:33Z" finished-at="2019-03-01T04:43:40Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at qa.sampleproject.utilities.util.takesScreenshotAtEndofTest(util.java:27)
at qa.sampleproject.utilities.WebEventLister.onException(WebEventLister.java:145)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$0(EventFiringWebDriver.java:84)
at com.sun.proxy.$Proxy7.onException(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$1(EventFiringWebDriver.java:107)
at com.sun.proxy.$Proxy8.quit(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.quit(EventFiringWebDriver.java:211)
at qa.sampleproject.tests.SampleTest.teardown(SampleTest.java:27)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:59)
at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:458)
at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:222)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:646)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- qa.sampleproject.tests.SampleTest -->
      <class name="qa.sampleproject.tests.yahooMailDatadriven">
        <test-method status="FAIL" signature="loginTest(java.lang.String, java.lang.String)[pri:3, instance:qa.sampleproject.tests.yahooMailDatadriven@4667ae56]" name="loginTest" duration-ms="0" started-at="2019-03-01T04:42:59Z" data-provider="getTestData1()" finished-at="2019-03-01T04:42:59Z">
          <exception class="org.testng.TestNGException">
            <message>
              <![CDATA[
Method public void qa.sampleproject.tests.yahooMailDatadriven.loginTest(java.lang.String,java.lang.String) requires a @DataProvider named : getTestData1()]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.TestNGException: 
Method public void qa.sampleproject.tests.yahooMailDatadriven.loginTest(java.lang.String,java.lang.String) requires a @DataProvider named : getTestData1()
at org.testng.internal.Parameters.findDataProvider(Parameters.java:513)
at org.testng.internal.Parameters.handleParameters(Parameters.java:691)
at org.testng.internal.ParameterHandler.handleParameters(ParameterHandler.java:49)
at org.testng.internal.ParameterHandler.createParameters(ParameterHandler.java:37)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:924)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.TestNGException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest -->
      </class> <!-- qa.sampleproject.tests.yahooMailDatadriven -->
      <class name="qa.sampleproject.tests.YahooPageTest">
        <test-method status="PASS" signature="setup()[pri:0, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="setup" is-config="true" duration-ms="10754" started-at="2019-03-01T04:42:34Z" finished-at="2019-03-01T04:42:45Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="PASS" signature="yahooLogoTest()[pri:1, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="yahooLogoTest" duration-ms="115" started-at="2019-03-01T04:42:45Z" finished-at="2019-03-01T04:42:45Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- yahooLogoTest -->
        <test-method status="PASS" signature="teardown()[pri:0, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="teardown" is-config="true" duration-ms="3664" started-at="2019-03-01T04:42:45Z" finished-at="2019-03-01T04:42:49Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method status="PASS" signature="setup()[pri:0, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="setup" is-config="true" duration-ms="8359" started-at="2019-03-01T04:42:49Z" finished-at="2019-03-01T04:42:57Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="FAIL" signature="yahooTitleTest()[pri:2, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="yahooTitleTest" duration-ms="31" started-at="2019-03-01T04:42:57Z" finished-at="2019-03-01T04:42:57Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [Yahoo - login] but found [Yahoo - login]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [Yahoo - login] but found [Yahoo - login]
at org.testng.Assert.fail(Assert.java:96)
at org.testng.Assert.failNotEquals(Assert.java:776)
at org.testng.Assert.assertEqualsImpl(Assert.java:137)
at org.testng.Assert.assertEquals(Assert.java:118)
at org.testng.Assert.assertEquals(Assert.java:453)
at org.testng.Assert.assertEquals(Assert.java:463)
at qa.sampleproject.tests.YahooPageTest.yahooTitleTest(YahooPageTest.java:33)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- yahooTitleTest -->
        <test-method status="PASS" signature="teardown()[pri:0, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="teardown" is-config="true" duration-ms="2284" started-at="2019-03-01T04:42:57Z" finished-at="2019-03-01T04:42:59Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method status="PASS" signature="setup()[pri:0, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="setup" is-config="true" duration-ms="12886" started-at="2019-03-01T04:42:59Z" finished-at="2019-03-01T04:43:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="FAIL" signature="loginTest()[pri:3, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="loginTest" duration-ms="25928" started-at="2019-03-01T04:43:12Z" finished-at="2019-03-01T04:43:38Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at qa.sampleproject.utilities.util.takesScreenshotAtEndofTest(util.java:27)
at qa.sampleproject.utilities.WebEventLister.onException(WebEventLister.java:145)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$0(EventFiringWebDriver.java:84)
at com.sun.proxy.$Proxy7.onException(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$1(EventFiringWebDriver.java:107)
at com.sun.proxy.$Proxy8.findElement(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.findElement(EventFiringWebDriver.java:194)
at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:69)
at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
at com.sun.proxy.$Proxy12.sendKeys(Unknown Source)
at qa.sampleproject.pages.yahoologinpage.yahoologin(yahoologinpage.java:32)
at qa.sampleproject.tests.YahooPageTest.loginTest(YahooPageTest.java:37)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- loginTest -->
        <test-method status="FAIL" signature="teardown()[pri:0, instance:qa.sampleproject.tests.YahooPageTest@1a968a59]" name="teardown" is-config="true" duration-ms="1689" started-at="2019-03-01T04:43:38Z" finished-at="2019-03-01T04:43:40Z">
          <exception class="java.lang.NullPointerException">
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException
at qa.sampleproject.utilities.util.takesScreenshotAtEndofTest(util.java:27)
at qa.sampleproject.utilities.WebEventLister.onException(WebEventLister.java:145)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$0(EventFiringWebDriver.java:84)
at com.sun.proxy.$Proxy7.onException(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.lambda$new$1(EventFiringWebDriver.java:107)
at com.sun.proxy.$Proxy8.quit(Unknown Source)
at org.openqa.selenium.support.events.EventFiringWebDriver.quit(EventFiringWebDriver.java:211)
at qa.sampleproject.tests.YahooPageTest.teardown(YahooPageTest.java:23)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:59)
at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:458)
at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:222)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:646)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- qa.sampleproject.tests.YahooPageTest -->
    </test> <!-- Test -->
  </suite> <!-- Suite -->
</testng-results>
